Userland/Libraries/LibGfx/ImageFormats/TGALoader.cpp

```diff
@@ -209,9 +209,9 @@ ErrorOr<bool> TGAImageDecoderPlugin::validate_before_create(ReadonlyBytes data)
     if (data.size() < sizeof(TGAHeader))
         return false;
     TGAHeader const& header = *reinterpret_cast<TGAHeader const*>(data.data());
-    if (header.data_type_code == TGADataType::UncompressedRGB && data.size() < static_cast<u64>(header.width) * header.height * (header.bits_per_pixel / 8))
```

Comment:
Can we factorize this code in something like `ensure_header_validity`?
Here we would have something like `return !ensure_header_validity(header).is_error()` and a simple `TRY()` in `decode_tga_header`